src/devdata/reset_modes.py

```diff
@@ -0,0 +1,132 @@
+        raise NotImplementedError
+
+    @property
+    @abc.abstractclassmethod
+    def description_for_confirmation(self) -> str:
+        raise NotImplementedError
+
+    def reset_database(self, django_dbname: str) -> None:
+        raise NotImplementedError
+
+    def __str__(self) -> str:
+        # Use the slug as the str for easier integration into the CLI
+        return self.slug
+
+
+class DropDatabaseReset(Reset):
+    """
+    Drop the entire database and re-create it using Django's test utils.
+
+    This is suitable in cases where Django is configured with a database
+    superuser account, which is likely to be the case in local development.
+    """
+
+    slug = "drop-database"
+
+    description_for_confirmation = "delete the database"
+
+    def reset_database(self, django_dbname: str) -> None:
+        db_conf = settings.DATABASES[django_dbname]
+        pg_dbname = db_conf["NAME"]
+
+        connection = connections[django_dbname]
+
+        with nodb_cursor(connection) as cursor:
+            cursor.execute("DROP DATABASE IF EXISTS {}".format(pg_dbname))
+
+            creator = connection.creation
+            creator._execute_create_test_db(
+                cursor,
+                {
+                    "dbname": pg_dbname,
+                    "suffix": creator.sql_table_creation_suffix(),
+                },
+            )
+
+
+class DropTablesReset(Reset):
+    """
+    Drop all the tables which Django knows about, including migration history.
+
+    This is suitable in cases where the current state of the database can be
+    assumed to be similar enough to the new state that removing the tables alone
+    is sufficient to clear out the data. For databases which support other forms
+    of data (e.g: Postgres sequences decoupled from tables) this mode will not
+    touch those data and the user must ensure they are handled suitably.
+
+    This is expected to be useful in cases where Django is configured with
+    administrative privileges within a database, but may not have access to drop
+    the entire database.
+
+    Note: this will not touch other database entities (e.g: Postgres sequences &
+    views) which may be present but are not managed by Django models -- even if
+    they were created by running migrations (e.g: via `RunSQL`).
+    """
+
+    slug = "drop-tables"
+
+    description_for_confirmation = "delete all tables in the database"
+
+    def reset_database(self, django_dbname: str) -> None:
+        connection = connections[django_dbname]
+
+        with connection.cursor() as cursor:
+            table_names = connection.introspection.table_names(cursor)
+
+            models = connection.introspection.installed_models(table_names)
+
+            if MigrationRecorder(connection).has_table():
+                models.add(MigrationRecorder.Migration)
+
+            with connection.schema_editor() as editor:
+                for model in models:
+                    editor.delete_model(model)
```

Comment:
Hrm, should this be transactional? That way we'd either drop all the tables or none, which is probably better than maybe dropping some?

A failure mode here is if the user has permission for some of the tables but not all.

Given that we're not dropping the database it might even be nice to have the whole import be transactional, but that likely requires deeper changes.